addons/payment/models/account_journal.py

```diff
@@ -11,46 +11,48 @@ class AccountJournal(models.Model):
     @api.constrains('inbound_payment_method_line_ids')
     def _check_inbound_payment_method_line_ids(self):
         """
-        Check and ensure that the user do not remove a apml that is linked to an acquirer in the test or enabled state.
+        Check and ensure that the user do not remove a apml that is linked to a provider in the test or enabled state.
         """
         if not self.company_id:
             return
 
         self.env['account.payment.method'].flush_model(['code', 'payment_type'])
         self.env['account.payment.method.line'].flush_model(['payment_method_id'])
-        self.env['payment.acquirer'].flush_model(['provider', 'state'])
+        self.env['payment.provider'].flush_model(['code', 'state'])
 
         self._cr.execute('''
-            SELECT acquirer.id
-            FROM payment_acquirer acquirer
-            JOIN account_payment_method apm ON apm.code = acquirer.provider
+            SELECT provider.id
+            FROM payment_provider provider
+            JOIN account_payment_method apm ON apm.code = provider.code
             LEFT JOIN account_payment_method_line apml ON apm.id = apml.payment_method_id AND apml.journal_id IS NOT NULL
-            WHERE acquirer.state IN ('enabled', 'test') AND apm.payment_type = 'inbound'
+            WHERE provider.state IN ('enabled', 'test') AND apm.payment_type = 'inbound'
             AND apml.id IS NULL
-            AND acquirer.company_id IN %(company_ids)s
+            AND provider.company_id IN %(company_ids)s
         ''', {'company_ids': tuple(self.company_id.ids)})
         ids = [r[0] for r in self._cr.fetchall()]
         if ids:
-            acquirers = self.env['payment.acquirer'].sudo().browse(ids)
-            raise UserError(_("You can't delete a payment method that is linked to an acquirer in the enabled or test state.\n"
-                              "Linked acquirer(s): %s", ', '.join(a.display_name for a in acquirers)))
+            providers = self.env['payment.provider'].sudo().browse(ids)
+            raise UserError(
+                _("You can't delete a payment method that is linked to a provider in the enabled or test state.\n"
+                  "Linked provider(s): %s", ', '.join(pro.display_name for pro in providers))
+            )
```

Comment:
```suggestion
            raise UserError(_("You can't delete a payment method that is linked to a provider in the enabled or test state.\n"
                              "Linked provider(s): %s", ', '.join(p.display_name for p in providers)))
```

No need to write `pro` instead of p here, it's clear enough with the `providers` variable. So this is a minimal diff. I usually whole fully agree to uniformized all payment files, but... well, review wise, it'll be easier for this task to keep the exact same line diff... :shrug: